packages/api/src/auth.js

```diff
@@ -30,7 +30,8 @@ export function withMagicToken (handler) {
     if (magicUser) {
       const userTags = await getUserTags(magicUser._id, env)
       request.auth = { user: magicUser, userTags }
-      env.sentry && env.sentry.setUser(magicUser)
+      // If env.log is not set, then the middlewares may be being run in the wrong order
+      env.log.setUser({ id: magicUser._id })
```

Comment:
Another thought that probably goes beyond this PR.
Setting the user is duplicated acroos different middlewares and I wonder if it could live in its own?

Reply:
I guess we could make it so that we always run the auth middleware first, and then the logging middleware afterwards, and then the logging middleware can know that the auth has already been done and so it can just look at `request.auth.user` to get the user. That way we wouldn't have to have the auth stuff calling the `setUser` method. But if we did it that way then:

* The auth middlewares wouldn't be able to do any logging (because the logging wouldn't yet be set up).
* We wouldn't be able to do `router.all('*', envAll)`, because we'd need that env stuff (or at least the logging part of it) to be run _after_ the auth middleware(s). So then I think our routing stuff would get quite complicated.

Maybe there's a third way, but having now thought that through, it makes me think that the current way is the least evil!

Reply:
For the record, we discussed this on Slack and concluded that it wasn't worth further discussion time for something so minor.